auth.php

```diff
@@ -47,6 +47,11 @@
 
 $currenturl = new moodle_url('/admin/tool/mfa/auth.php');
 
+// Check if enough factors have been passed from no input factors.
+if (tool_mfa_user_passed_enough_factors()) {
+    $SESSION->tool_mfa_authenticated = true;
```

Comment:
This looks like we are duplicating, but only partially, logic which is here:

https://github.com/catalyst/moodle-tool_mfa/blob/master/auth.php#L95-L107

Perhaps refactor both of those into a new manager method?